### PR TITLE
added log-cauchy distribution

### PR DESCRIPTION
**Pull request by mayankray2020**
* stats
  * implemented LogCauchy Distribution

### CODE REVIEW

File: sympy/stats/crv_types.py [hunk 1]
```diff
@@ -2279,6 +2281,76 @@ def Levy(name, mu, c):
...
+    >>> density(X)(z)
+    1/(5*pi*z*((log(z) - 2)**2 + 1/25))
+
+    >>> cdf(X)(z)
+    atan(5*log(z) - 10)/pi + 1/2
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/Log-Cauchy_distribution
+    """
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py:2350]:
```suggestion
    r"""
   Create a continuous random variable with a Log-Cauchy distribution.
    The density of the Log-Cauchy distribution is given by

    .. math::
        f(x) := \frac{1}{\pi x} \frac{\gamma}{(log(x)-\x0^2) + \gamma^2}

    Parameters
    ==========

    x0 : Real number, the location

    gamma : Real number, `\gamma > 0`, a scale

    Returns
    =======

    RandomSymbol

    Examples
    ========

    >>> from sympy.stats import  LogCauchy, density, cdf
    >>> from sympy import Symbol, S

    >>> x0 = 2
    >>> gamma = S.One / 5
    >>> z = Symbol("z")

    >>> X = LogCauchy("x", x0, gamma)

    >>> density(X)(z)
    1/(5*pi*z*((log(z) - 2)**2 + 1/25))

    >>> cdf(X)(z)
    atan(5*log(z) - 10)/pi + 1/2

    References
    =========

    .. [1] https://en.wikipedia.org/wiki/Log-Cauchy_distribution
    """
```

File: sympy/stats/tests/test_continuous_rv.py [hunk 2]
```diff
@@ -842,6 +842,16 @@ def test_levy():
...
     raises(ValueError, lambda: Levy('x',mu,c))
 
+def test_logcauchy():
+    x0 = 2
+    gamma = S.One /5
+
+    X = LogCauchy("x", x0, gamma)
+    #Tests density_funtion
+    assert density(X)(2) - 1/(10*pi*(1/25 + (-2 + log(2))**2)) < 0.001
+    assert cdf(X)(2) - (-atan(10 - 5*log(2))/pi + 1/2) < 0.001
```
Comment (czgdp1807) [inline sympy/stats/tests/test_continuous_rv.py:852]: Please add symbolic tests instead.

File: sympy/stats/crv_types.py (hunk 1 above)
Comment (mayankray2020) [reply sympy/stats/crv_types.py]: done

File: sympy/stats/tests/test_continuous_rv.py (hunk 2 above)
Comment (mayankray2020) [reply sympy/stats/tests/test_continuous_rv.py]: added symbolic tests

File: sympy/stats/crv_types.py [hunk 3]
```diff
@@ -2279,6 +2281,85 @@ def Levy(name, mu, c):
...
+                                  "Log-Cauchy distribution does not exist.")
+
+    def _moment_generating_function(self, t):
+        raise NotImplementedError("The moment generating function for the "
+                                  "Log-Cauchy distribution does not exist.")
+
+def LogCauchy(name, x0, gamma):
+    r"""
+   Create a continuous random variable with a Log-Cauchy distribution.
+    The density of the Log-Cauchy distribution is given by
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py:2320]:
```suggestion
    Create a continuous random variable with a Log-Cauchy distribution.
    The density of the Log-Cauchy distribution is given by
```

File: sympy/stats/crv_types.py [hunk 4]
```diff
@@ -2279,6 +2281,85 @@ def Levy(name, mu, c):
...
+    atan(5*log(z) - 10)/pi + 1/2
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Log-Cauchy_distribution
+    """
+
+    return rv(name, LogCauchyDistribution, (x0, gamma))
+#-------------------------------------------------------------------------------
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py:2362]:
```suggestion
    return rv(name, LogCauchyDistribution, (x0, gamma))


#-------------------------------------------------------------------------------
```

File: sympy/stats/crv_types.py [hunk 5]
```diff
@@ -2279,6 +2281,85 @@ def Levy(name, mu, c):
     return rv(name, LevyDistribution, (mu, c))
 
 #-------------------------------------------------------------------------------
+# Log-Cauchy distribution --------------------------------------------------------
+
+
+class LogCauchyDistribution(SingleContinuousDistribution):
+    _argnames = ('x0', 'gamma')
+
+    set = Interval(0, oo)
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py:2290]:
```suggestion
    set = Interval.open(0, oo)
```

File: sympy/stats/crv_types.py [hunk 6]
```diff
@@ -2279,6 +2281,85 @@ def Levy(name, mu, c):
...
+
+    @property
+    def set(self):
+        return Interval(self.x0, self.b)
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py:2294]:
```suggestion
```

File: sympy/stats/crv_types.py [hunk 7]
```diff
@@ -2279,6 +2281,85 @@ def Levy(name, mu, c):
...
+    >>> density(X)(z)
+    1/(5*pi*z*((log(z) - 2)**2 + 1/25))
+
+    >>> cdf(X)(z)
+    atan(5*log(z) - 10)/pi + 1/2
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Log-Cauchy_distribution
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py:2358]: Please use a reference where `gamma`, `x0` notation is used. In the above link, they use, `mu` and `sigma`.

File: sympy/stats/crv_types.py (hunk 4 above)
Comment (mayankray2020) [reply sympy/stats/crv_types.py]: done

File: sympy/stats/crv_types.py (hunk 5 above)
Comment (mayankray2020) [reply sympy/stats/crv_types.py]: done

File: sympy/stats/crv_types.py (hunk 6 above)
Comment (mayankray2020) [reply sympy/stats/crv_types.py]: done

File: sympy/stats/crv_types.py (hunk 7 above)
Comment (mayankray2020) [reply sympy/stats/crv_types.py:2354]: used mu and sigma notation

**Comment by mayankray2020** [comment]
@czgdp1807 please review this

File: sympy/stats/crv_types.py [hunk 8]
```diff
@@ -2280,6 +2282,83 @@ def Levy(name, mu, c):
...
+
+class LogCauchyDistribution(SingleContinuousDistribution):
+    _argnames = ('mu', 'sigma')
+
+    set = Interval.open(0, oo)
+
+    @staticmethod
+    def check(mu, sigma):
+        _value_check(sigma > 0, "Scale parameter Gamma must be positive.")
+        _value_check(mu.is_real, "Location parameter must be real.")
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py:2297]:
```suggestion
        _value_check(mu.is_real != False, "Location parameter must be real.")
```

File: sympy/stats/crv_types.py [hunk 9]
```diff
@@ -2280,6 +2282,83 @@ def Levy(name, mu, c):
...
+
+
+class LogCauchyDistribution(SingleContinuousDistribution):
+    _argnames = ('mu', 'sigma')
+
+    set = Interval.open(0, oo)
+
+    @staticmethod
+    def check(mu, sigma):
+        _value_check(sigma > 0, "Scale parameter Gamma must be positive.")
```
Comment (czgdp1807) [inline sympy/stats/crv_types.py:2296]:
```suggestion
        _value_check((sigma > 0) != False, "Scale parameter Gamma must be positive.")
```

Comment (mayankray2020) [reply sympy/stats/crv_types.py]: done @czgdp1807

**Comment by czgdp1807** [comment]
LGTM. Will merge tomorrow if no objections raised.

**Pull request merged**